flask-server/tests/test_forum_api.py

```diff
@@ -0,0 +1,128 @@
+import unittest
+from test_api import TestAPI
+
+
+class TestForumAPI(TestAPI):
+    """Tests for forum namespace routes"""
+
+    def test_get_all_posts(self):
+        """Test getting all forum posts"""
+        forum_response = self.client.get("/forum/all")
+
+        status_code = forum_response.status_code
+
+        self.assertEqual(status_code, 200)
+
+    def test_get_post_id_not_found(self):
+        """Test get post by id route, with no data"""
+        id=1
+        response = self.client.get(f"/forum/id/{id}")
+        status_code = response.status_code
+
+        self.assertEqual(status_code, 404)
+
+    def test_create_post_successful(self):
+        """Test creating a post, login required for @jwt_required route"""
+        register_response = self.client.post("/user/register",
+            json = {
+                "username": "testuser",
+                "first_name": "test",
+                "last_name": "user",
+                "email": "testuser@test.com",
+                "password": "mytestpassword"
+            }
+        )
```

Comment:
I can imagine there is a "beforeEach" like function with pytesting too. It'd be code that is executed before every single test. THat would be useful to move this user creation to, since otherwise you will be repeating this code a lot.

If thats not possible, you could make a create_user() function to do this and call that (or like a setup() function) at the start of the relevant tests, just to clean this up a bit. 

Once you get more complex cases these can get very finicky.

Reply:
Yeah that would be very useful, I'll look into.

Reply:
Ah I think the beforeEach equivalent is setUp, and the end of test equivalent is tearDown. For mine I have the table creation. Think you can add whatever you like but to test every different scenario probably better I don't default mock sign up members. So being smarter with mocking or seeing if there are any tools unittest has for it would be the way forward I think.

Reply:
To reduce repetition I have made a function in the main test API file to create a default user and updated the existing tests to use it
75faf62
Will add more for other cases that have duplication

Reply:
Testing "complete"
56f808b
Rough summary of tests and associated functions attached
![image](https://github.com/Angel2001-programmer/Introverse/assets/130995689/222d9123-3f6a-4fd0-b2a7-b4620d301611)